src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -538,8 +538,47 @@
 
    if (length(pieces) > 1)
       print(help(pieces[2], package=pieces[1], help_type='html'))
-   else
-      print(help(pieces[1], help_type='html', try.all.packages=T))
+   else {
+      # try custom help handler, otherwise fall through to default handler
+      helpUrl <- .rs.getCustomHelpUrl(token)
+      if (!is.null(helpUrl))
+         if (nzchar(helpUrl)) # handlers return "" to indicate no help available
+            utils::browseURL(helpUrl)
+      else
```

Comment:
Yikes! Just pushed a fix for this.

On Fri, Sep 9, 2016 at 2:39 PM, Kevin Ushey <notifications@github.com>
wrote:

> In src/cpp/session/modules/SessionCodeTools.R
> <https://github.com/rstudio/rstudio/pull/780#discussion_r78228805>:
>
> > @@ -538,8 +538,47 @@
> >
> >     if (length(pieces) > 1)
> >        print(help(pieces[2], package=pieces[1], help_type='html'))
> > -   else
> > -      print(help(pieces[1], help_type='html', try.all.packages=T))
> > +   else {
> > +      # try custom help handler, otherwise fall through to default handler
> > +      helpUrl <- .rs.getCustomHelpUrl(token)
> > +      if (!is.null(helpUrl))
> > +         if (nzchar(helpUrl)) # handlers return "" to indicate no help available
> > +            utils::browseURL(helpUrl)
> > +      else
>
> Is this misleading indentation here? (nested unbraced if-else)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rstudio/rstudio/pull/780/files/4fb3ff12cf75dcd139ad454c3cccd497e10af235#r78228805>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGXx_239iQoNgy3OTkoCcWVaBQ3o0a3ks5qoafugaJpZM4J5Tzb>
> .
>